src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,46 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int totalBuy = 0;
+        int totalSupply = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");//podział na typ i wartość
+                String type = parts[0]; // buy-supply
+                int value = Integer.parseInt(parts[1]);//wartość
+
+                if ("buy".equalsIgnoreCase(type)) {
+                    totalBuy += value;
+                } else if ("supply".equalsIgnoreCase(type)) {
+                    totalSupply += value;
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("Error reading file: " + e.getMessage());
+            return;
+        } catch (NumberFormatException e) {
+            System.out.println("Invalid number format: " + e.getMessage());
```

Instead of printing the exception message, consider re-throwing a `RuntimeException` with a meaningful message. This aligns with the checklist item: 'Don't ignore exceptions'.